src/TextInput2/TextInput2.tsx

```diff
@@ -83,7 +91,7 @@ const TextInput2: React.FC<Props> = ({
 
   useEffect(() => {
```

Comment:
After reading about some best practices for error messaging, we can probably rewrite this entire thing. Something like this maybe:
```
useEffect(() => {
  // if input is empty and requirement error is active, then do nothing and return
  // if input is empty, then clear error and return


  // check error validation function
  // if no error validation, clear error and return
  // if errormessage exists and is not focused, then set error and return
});
```